easypy/randutils.py

```diff
@@ -42,6 +47,49 @@ def get_chunks(self, offset, end, block_size_range, shuffle=False):
             offset += size
 
 
+def load_dictionary(filename):
+    full_path = os.path.join(os.path.dirname(__file__), 'dictionary', filename)
+    with open(full_path) as dict_file:
+        dictionary = [line.strip() for line in dict_file]
+
+    grouped_by_len = grouped(dictionary, key=len)
+    return collections.defaultdict(list, grouped_by_len)
+
+
+NOUNS = load_dictionary('nouns.txt')
+VERBS = load_dictionary('verbs.txt')
+ADVERBS = load_dictionary('adverbs.txt')
+ADJECTIVES = load_dictionary('adjectives.txt')
+PARTS_OF_SPEECH = (NOUNS, VERBS, ADVERBS, ADJECTIVES)
+
+
+def random_meaningful_name(max_length=50, sep='-'):
+    parts = []
+    remaining = max_length
+
+    if max_length < 3:
+        return random_string(max_length, charset=string.ascii_lowercase)
+
+    for part_group in PARTS_OF_SPEECH:
+        max_part_len = remaining - len(sep) if parts else max_length
```

Comment:
That's why I have `if parts else max_length` that way I am always calculating the prefixing separator. i.e. I do not add it for the first word.